lib/compiler/src/engine/artifact.rs

```diff
@@ -79,7 +90,18 @@ impl Artifact {
             table_styles,
         )?;
 
-        Self::from_parts(&mut inner_engine, artifact)
+        Ok(artifact)
+    }
+
+    /// Compile a data buffer into a `ArtifactBuild`, which may then be instantiated.
+    #[cfg(feature = "compiler")]
+    pub fn get_module_info(
+        engine: &Engine,
+        data: &[u8],
+        tunables: &dyn Tunables,
+    ) -> Result<ModuleInfo, CompileError> {
+        let artifact = Self::get_artifact_build(engine, data, tunables)?;
+        Ok(artifact.create_module_info())
     }
```

Comment:
Why don't we move this into the `Engine`, so we have in the `Engine` impl:

```rust
    #[cfg(feature = "compiler")]
    pub fn get_module_info(
        data: &[u8],
    ) -> Result<ModuleInfo, CompileError> {
        // this is from `artifact_builder.rs`
        let environ = ModuleEnvironment::new();
        let translation = environ.translate(data).map_err(CompileError::Wasm)?;
        return translation.module;
    }
```

Reply:
Done in https://github.com/wasmerio/wasmer/pull/3483/commits/b26b48794cc7b00bb5f939dfdf40bba94e9ba86c